deepchem/utils/poly_wd_graph_utils.py

```diff
@@ -0,0 +1,478 @@
+    encoding = [0] * (len(choices) + 1)
+    index = choices.index(value) if value in choices else -1
+    encoding[index] = 1
+
+    return encoding
+
+
+def generate_atom_features(atom: Chem.rdchem.Atom,
+                           PARAMS: FeaturizationParameters,
+                           functional_groups=None):
+    """
+    This function generates the feature vector for an atom.
+
+    Parameters
+    ----------
+    atom : rdkit.Chem.rdchem.Atom
+        RDKit atom object.
+    PARAMS : deepchem.feat.FeaturizationParameters
+        Featurization parameters.
+    functional_groups : list, optional
+        List of functional groups.
+
+    Returns
+    -------
+    list
+        Feature vector.
+
+    Examples
+    --------
+    >>> import deepchem as dc
+    >>> mol = Chem.MolFromSmiles("C")
+    >>> PARAMS = dc.feat.FeaturizationParameters()
+    >>> for atom in mol.GetAtoms():
+    ...     atom_feat_vector = dc.utils.generate_atom_features(
+    ...         atom, PARAMS = PARAMS))
+    [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0,
+    1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0.12011]
+    """
+    if atom is None:
+        features = [0] * PARAMS.ATOM_FDIM
+    else:
+        features = onek_encoding_unk(atom.GetAtomicNum(
+        ) - 1, PARAMS.ATOM_FEATURES['atomic_num']) + onek_encoding_unk(
+            atom.GetTotalDegree(),
+            PARAMS.ATOM_FEATURES['degree']) + onek_encoding_unk(
+                atom.GetFormalCharge(),
+                PARAMS.ATOM_FEATURES['formal_charge']) + onek_encoding_unk(
+                    int(atom.GetChiralTag()),
+                    PARAMS.ATOM_FEATURES['chiral_tag']) + onek_encoding_unk(
+                        int(atom.GetTotalNumHs()),
+                        PARAMS.ATOM_FEATURES['num_Hs']) + onek_encoding_unk(
+                            int(atom.GetHybridization()),
+                            PARAMS.ATOM_FEATURES['hybridization']
+                        ) + [1 if atom.GetIsAromatic() else 0] + [
+                            atom.GetMass() * 0.01
+                        ]  # scaled to about the same range as other features
+        if functional_groups is not None:
+            features += functional_groups
+    return features
+
+
+def generate_bond_features(bond: Chem.rdchem.Bond,
+                           PARAMS: FeaturizationParameters):
+    """
+    This function generates the feature vector for a bond.
+
+    Parameters
+    ----------
+    bond : rdkit.Chem.rdchem.Bond
+        RDKit bond object.
+    PARAMS : deepchem.feat.FeaturizationParameters
+        Featurization parameters.
+
+    Returns
+    -------
+    list
+        Feature vector.
+
+    Examples
```

Comment:
Pls keep examples before Parameters and Returns

Reply:
Will do